compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaErrorTypeSymbol.java

```diff
@@ -90,7 +90,10 @@ public String signature() {
             return this.signature;
         }
 
-        String definitionName = getBType().tsymbol.name.value;
+        String definitionName = "error";
+        if (detailTypeDescriptor().typeKind() == TypeDescKind.TYPE_REFERENCE) {
```

Comment:
This is redundant and restrictive. e.g., `error<record {| int code; |}>` We just simply need the detail type's signature.

Reply:
Placed an additional check to handle this.